torch/csrc/jit/ir.h

```diff
@@ -60,6 +60,15 @@ static inline bool operator==(const Use & a, const Use & b) {
 // Graph holds a list of parameters.
 struct Param;
 
+// SourceLocation represents source code-level debug information for a node.
+// It contains a Python stack trace that represents the provenance of a given
+// node in the trace.
+struct SourceLocation {
+  SourceLocation(std::string python_traceback)
```

Comment:
nit: `const std::string &` to avoid 2 copies

Reply:
I did this as a value since I expect this constructor to be called with an rvalue argument, so the actual operation is going to be 1 move out of that object. I looked through the compiler output and that seems to be what's happening, in fact it moved directly out of the tail call in `getPythonInterpreterStackTrace` into the member. Thoughts?

Reply:
It's better to take it as a value because you're going to copy it anyway. Just remember to `std::move` it when constructing the member